### PR TITLE
SONAR-6716 remove data migrations of v4.4

### PR DESCRIPTION
**Pull request by simonbrandhof**
SONAR-6716 remove data migrations of v4.4

### CODE REVIEW

File: sonar-db/src/main/java/org/sonar/db/version/v50/RemoveSortFieldFromIssueFiltersMigrationStep.java [hunk 1]
```diff
@@ -17,21 +17,19 @@
...
-import org.sonar.server.db.migrations.Select;
-import org.sonar.server.db.migrations.SqlStatement;
-
-import com.google.common.collect.Lists;
+import org.sonar.db.version.BaseDataChange;
+import org.sonar.db.version.MassUpdate;
+import org.sonar.db.version.Select;
+import org.sonar.db.version.SqlStatement;
 
 // TODO could be refactored to benefit from existing code of ReplaceIssueFiltersProjectKeyByUuid
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/version/v50/RemoveSortFieldFromIssueFiltersMigrationStep.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

File: sonar-db/src/main/java/org/sonar/db/version/Select.java [hunk 2]
```diff
@@ -25,7 +25,6 @@
 import java.sql.Timestamp;
 import java.util.Date;
 import java.util.List;
-
 import javax.annotation.CheckForNull;
 
 public interface Select extends SqlStatement<Select> {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/version/Select.java]: MINOR Move constants to a class or enum. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1214)

**Comment by SonarTech** [comment]
SonarQube analysis reported 38 issues:
* CRITICAL 6 critical
* MAJOR 12 major
* MINOR 18 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* CRITICAL [Either log or rethrow this exception.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/Select.java#L123) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* INFO [Complete the task associated to this TODO comment.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/SelectImpl.java#L93) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* MAJOR [Make "newExceptionWithRowDetails" a "static" method.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/SelectImpl.java#L88) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2325)
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v451/DeleteUnescapedActivities.java#L47) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v50/FeedFileSources.java#L232) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v50/FeedIssueLongDates.java#L48) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)
* MAJOR [Reduce this anonymous class number of lines from 21 to at most 20, or make it a named class.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v50/FeedIssueLongDates.java#L48) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1188)
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v50/FeedSnapshotSourcesUpdatedAt.java#L51) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)
* CRITICAL [Use a stronger encryption algorithm than MD5.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v50/FileSourceDto.java#L114) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2070)
* MAJOR [Constructor has 14 parameters, which is greater than 7 authorized.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v50/FileSourceDto.java#L66) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)
* CRITICAL [Either log or rethrow this exception.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v50/FileSourceDto.java#L183) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* CRITICAL [Format specifiers should be used instead of string concatenation.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/AddNewCharacteristics.java#L233) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2275)
* CRITICAL [Format specifiers should be used instead of string concatenation.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/AddNewCharacteristics.java#L244) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2275)
* CRITICAL [Format specifiers should be used instead of string concatenation.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/AddNewCharacteristics.java#L246) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2275)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L80) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L136) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MAJOR [The Cyclomatic Complexity of this method "toBinary" is 23 which is greater than 10 authorized.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L65) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
* MAJOR [Make "toBinary" a "static" method.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L65) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2325)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L124) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L112) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L84) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L100) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L92) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L156) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L120) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L116) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L140) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L88) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L104) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L132) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L128) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L96) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L108) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedIssueTags.java#L54) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)
* MINOR [Split this 182 characters long line (which is greater than 180 authorized).](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v51/FeedSnapshotsLongDates.java#L46) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonarqube/blob/1da81855213565d193a495ae0f261985b3fc0568/sonar-db/src/main/java/org/sonar/db/version/v52/MoveProjectProfileAssociation.java#L123) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)

**Comment by julienlancelot** [comment]
You can also remove CopyRequirementsFromCharacteristicsToRules, as it as been introduced in 4.3, so it should not be needed anymore.

**Comment by julienlancelot** [comment]
When starting from a not compatible previous version (for instance 4.4), server startup fail with a NPE

**Comment by julienlancelot** [comment]
In DropRuleTagsTables, I think you could remove migration of data (from line 39 to 72)

**Comment by julienlancelot** [comment]
You can remove migration UpdateMeasureFiltersOnLanguage

**Comment by julienlancelot** [comment]
Apply my feedback, and LGTM !

**Comment by SonarTech** [comment]
SonarQube analysis reported 38 issues:
* CRITICAL 6 critical
* MAJOR 12 major
* MINOR 18 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* CRITICAL [Either log or rethrow this exception.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/Select.java#L123) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Make "newExceptionWithRowDetails" a "static" method.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/SelectImpl.java#L88) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2325)
* INFO [Complete the task associated to this TODO comment.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/SelectImpl.java#L93) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v451/DeleteUnescapedActivities.java#L47) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v50/FeedFileSources.java#L232) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)
* MAJOR [Reduce this anonymous class number of lines from 21 to at most 20, or make it a named class.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v50/FeedIssueLongDates.java#L48) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1188)
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v50/FeedIssueLongDates.java#L48) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v50/FeedSnapshotSourcesUpdatedAt.java#L51) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)
* CRITICAL [Either log or rethrow this exception.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v50/FileSourceDto.java#L183) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Constructor has 14 parameters, which is greater than 7 authorized.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v50/FileSourceDto.java#L66) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)
* CRITICAL [Use a stronger encryption algorithm than MD5.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v50/FileSourceDto.java#L114) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2070)
* CRITICAL [Format specifiers should be used instead of string concatenation.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/AddNewCharacteristics.java#L233) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2275)
* CRITICAL [Format specifiers should be used instead of string concatenation.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/AddNewCharacteristics.java#L246) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2275)
* CRITICAL [Format specifiers should be used instead of string concatenation.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/AddNewCharacteristics.java#L244) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2275)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L80) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L120) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L112) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L116) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L84) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MAJOR [Make "toBinary" a "static" method.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L65) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2325)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L104) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L124) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L136) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L156) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L100) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MAJOR [The Cyclomatic Complexity of this method "toBinary" is 23 which is greater than 10 authorized.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L65) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L128) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L92) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L108) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L88) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L140) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L132) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MINOR [Refactor this code to not nest more than 4 if/for/while/switch/try statements.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java#L96) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedIssueTags.java#L54) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)
* MINOR [Split this 182 characters long line (which is greater than 180 authorized).](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v51/FeedSnapshotsLongDates.java#L46) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonarqube/blob/da39d9ab8746bd9aecd4d41cf83d0269c8fd835d/sonar-db/src/main/java/org/sonar/db/version/v52/MoveProjectProfileAssociation.java#L123) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)

**Pull request closed without merging**